### PR TITLE
Update libgit2 to 9bbc8f3

### PR DESCRIPTION
**Pull request by nulltoken**
https://github.com/libgit2/libgit2/compare/e0902fb...bdf0e73

### CODE REVIEW

**Comment by nulltoken** [comment]
This is an attempt at troubleshooting #961 build issues on Mac Os X

**Comment by nulltoken** [comment]
:crying_cat_face: 

@carlosmn @Therzok Any idea? Can you reproduce this locally?

**Comment by Therzok** [comment]
Yep.

**Comment by nulltoken** [comment]
I'll work on bisecting this.

**Comment by nulltoken** [comment]
Mac Os X build passes against libgit2/libgit2@b23c206
Mac Os X build fails against libgit2/libgit2@a291790
 
So this issue lies somewhere in https://github.com/libgit2/libgit2/compare/b23c206...a291790

As shown in https://github.com/libgit2/libgit2/pull/2917, the issue is put under the light through the  `-DCMAKE_OSX_ARCHITECTURES=i386;x86_64` option.

/cc @jeffhostetler @ethomson @carlosmn @Therzork Any idea?

**Comment by nulltoken** [comment]
Updated to target @jeffhostetler 's https://github.com/libgit2/libgit2/pull/2919

File: LibGit2Sharp/Core/GitCheckoutOpts.cs [hunk 1]
```diff
@@ -95,6 +96,18 @@ internal enum CheckoutStrategy
         GIT_CHECKOUT_CONFLICT_STYLE_DIFF3 = (1 << 21),
 
         /// <summary>
+ 	    /// Don't overwrite existing files or folders
+        /// </summary>
+ 	    GIT_CHECKOUT_DONT_REMOVE_EXISTING = (1 << 22),
```
Comment (Therzok) [inline LibGit2Sharp/Core/GitCheckoutOpts.cs]: ZOMG TABS. :trollface:

Comment (nulltoken) [inline LibGit2Sharp/Core/GitCheckoutOpts.cs]: FIXED :wink:

**Comment by nulltoken** [comment]
Ok. Now that builds locally. Let's see how many tests are broken now...

**Comment by nulltoken** [comment]
FWIW, there are a **ton** of breaking changes (and I see no easy way of providing a deprecating path that would actually be useful)...

**Comment by nulltoken** [comment]
Regarding the identity which is going to be used for reflog messages, I was thinking about adding an entry to `RepositoryOptions` and `CloneOptions`.

Thoughts?

**Comment by nulltoken** [comment]
Ok. This is ugly. But.. this is the first build that passes!

@carlosmn @ethomson @jamill @Therzok Would you please take a look at this and provide me with an early feedback?

**Comment by Therzok** [comment]
o-shi---

Grabbing coffee.

**Comment by Therzok** [comment]
I have a few things I want to put in for discussion:
 * Signature should have a constructor that takes an identity and uses DateTimeOffset.Now as time.
 * We should enforce Identity in the RepositoryOptions constructor.

More to come.

**Comment by carlosmn** [comment]
It looks good to me as far as the delegation to libgit2 goes, though I notice the methods in `Proxy` do not have the `_from_` infix for creating from an annotated commit.

The mess with `UpdateHeadTarget()` seems to stem from the special-casing you have for "HEAD", allowing it to point to an arbitrary reference, whereas no other symref is allowed to do that. Personally I would favour not checking for the existence of the target. If a caller wants to make sure it exists, they can look it up and pass the `Reference` themselves (not that this really solves anything, as it's still subject to race conditions on the target existing).

The current scheme may already be broken for allowing a remote-tracking HEAD to point to a branch which you know is coming.

**Comment by nulltoken** [comment]
I've extracted some fixes into #997

A reduced diff can be examined at https://github.com/libgit2/libgit2sharp/compare/ntk/deprecate...ntk/upgrade_binaries

**Comment by nulltoken** [comment]
>Signature should have a constructor that takes an identity and uses DateTimeOffset.Now as time.

@Therzok Fixed

>We should enforce Identity in the RepositoryOptions constructor.

Meaning, make it mandatory?

**Comment by nulltoken** [comment]
>It looks good to me as far as the delegation to libgit2 goes, though I notice the methods in Proxy do not have the _from_ infix for creating from an annotated commit.

@carlosmn Fixed

**Comment by Therzok** [comment]
I'm not sure, but from my experience, not setting an Identity/Signature failed to commit. (even though I had a global config set up)

With this Identity class, we can probably set up just the user/email and make lg2s do the time resolution based on the actual system time.

Just my $0.02. I'm not sure this would actually work every time.

**Comment by nulltoken** [comment]
>I'm not sure, but from my experience, not setting an Identity/Signature failed to commit. (even though I had a global config set up)

@Therzok That's really weird. Could you please try to check this against this version?

**Comment by nulltoken** [comment]
FWIW, I've cleaned up the PR and rebased it.

Final comments before merge?

**Comment by Therzok** [comment]
When I get back to redoing some bits of the libgit2 port of XS, I'll take a look at it again. Until then, stay tuned.

**Comment by nulltoken** [comment]
Published as a NuGet pre-relase package `LibGit2Sharp 0.22.0-pre20150317225305`

**Pull request merged**